### PR TITLE
Kubernetes Operator Reference

### PR DESCRIPTION
**Pull request by ravindk89**
Done:
- Skeleton structure + Intro
- CRD Documentation

Strongly recommend building from source and using the HTML output side-by-side with the RAW to better understand how some of the linking works out.

### CODE REVIEW

File: source/kubernetes/minio-operator-reference.rst [hunk 1]
```diff
@@ -104,94 +127,689 @@ The MinIO Operator adds a
...
+   - ``types`` specifies ``kubernetes.io/tls``
+
+   .. code-block:: yaml
+
+      spec:
+         externalCertSecret:
+            - name: tenant-external-cert-secret-name
+              type: kubernetes.io/tls
+
+.. kubeconf:: spec.externalClientCertSecret
```
Comment (ravindk89) [inline source/kubernetes/minio-operator-reference.rst:519]: @nitisht Is there a reason to not also have an array here vs an object? A MinIO Tenant could have clients using multiple custom CAs right?

File: source/kubernetes/minio-operator-reference.rst [hunk 2]
```diff
@@ -104,94 +127,689 @@ The MinIO Operator adds a
...
+   *Required*
+
+   The configuration for each MinIO Zone deployed in the MinIO Tenant.
+
+
+
+
+     
+
+Transport Layer Encryption (TLS)
```
Comment (ravindk89) [inline source/kubernetes/minio-operator-reference.rst:442]: @nitisht do any of the certificate related fields have any effect if `spec.autoCert` is 'disabled' or omitted?

**Comment by ravindk89** [comment]
@nitisht @krisis please take a look when you can.If you need to punt please let me know who best can take a look.

File: source/kubernetes/minio-operator-reference.rst (hunk 1 above)
Comment (nitisht) [reply source/kubernetes/minio-operator-reference.rst:810]: No reason as such. May be we can make it an array going forward.

**Review by nitisht — CHANGES_REQUESTED** [review]
one small nit, otherwise LGTM

**Review by krisis — APPROVED** [review]

File: source/kubernetes/minio-operator-reference.rst [hunk 3]
```diff
@@ -27,10 +27,33 @@ and managing MinIO Tenants.
...
+
+      Use :github:`kustomize <kubernetes-sigs/kustomize>` to deploy the
+      MinIO Operator:
+
+      .. code-block::
+         :class: copyable
+         :substitutions:
+
+         kustomize build github.com/minio/operator/\?ref\=|minio-operator-latest-version| \
+            > minio-operator-|minio-operator-latest-version|.yaml
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/kubernetes/minio-operator-reference.rst:54]: Made a tweak.

**Pull request merged**